.github/workflows/validate.yml

```diff
@@ -1,7 +1,12 @@
 ---
-# These jobs are run anytime a PR is opened against the main branch. In brief, to get a PR merged, you must:
-#   - Update the version number in pyproject.toml
-#   - Not have any linting issues (run `ruff format`)
+# The general flow of this is like so:
+#
+# - When a PR is opened against `main`, or if that PR changes:
+#   - Figure out if any code changes have been made
+#   - Require a version number change
```

Comment:
I'm not sure if we want this all the time; if we start a new project and want to stay in 'dev' mode on main until we release, then it would be meaningless to increment some artificial version number. IMO, if we want to make versioned releases, we should cut new branches with that intent.

Reply:
oh, or is this just for this repo's workflow, and not part of the pulumi shared workflow code?

Reply:
This workflow is designed for this project's workflow only. This concerns how this project's PRs are processed.